plugins/source/azuredevops/client/client.go

```diff
@@ -2,24 +2,45 @@ package client
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
-type AzureDevOpsClient struct {
+type Client struct {
+	logger     zerolog.Logger
+	Connection *azuredevops.Connection
 }
 
-func (*AzureDevOpsClient) Logger() *zerolog.Logger {
+func (*Client) Logger() *zerolog.Logger {
 	return &log.Logger
 }
 
-func (*AzureDevOpsClient) ID() string {
+func (*Client) ID() string {
 	return "AzureDevOpsClient"
 }
 
 func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.ClientMeta, error) {
-	return &AzureDevOpsClient{}, nil
+	var spec Spec
+	err := s.UnmarshalSpec(&spec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal GitHub spec: %w", err)
+	}
+
+	if spec.PersonalAccessToken == "" {
```

Comment:
I really think we should allow for env variables for keys.
@jsonpr could you share any insights on the security perspective for the keys hard-coded in the config?